components/raftstore-v2/src/tablet.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::sync::{
+    atomic::{AtomicU64, Ordering},
+    Arc, Mutex,
+};
+
+struct LatestTablet<EK> {
+    data: Mutex<Option<EK>>,
+    version: AtomicU64,
+}
+
+/// Tablet may change during split, merge and applying snapshot. So we need a shared value to
+/// reflect the latest tablet. `CachedTablet` provide cache that can speed up common access.
+#[derive(Clone)]
+pub struct CachedTablet<EK> {
```

Comment:
tablet is already cached in TabletFactoryV2. And we can add a method in TabletFactoryV2 to get latest tablet by region id.   I think having all cached tablet in TabletFactory makes it more clear.

Reply:
No, they are different. `CachedTablet` is supposed to be shared between `Peer`, `Apply` and `ReadDelegate`. Using `TabletFactoryV2` is a bad idea as it needs to lock and hash get for every reads/writes.